testnet/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -0,0 +1,699 @@
+        }
+    }
+
+    fn submit_operation(&mut self, operation: BurnchainOperationType, op_signer: &mut BurnchainOpSigner) {
+
+        let transaction = match operation {
+            BurnchainOperationType::LeaderBlockCommit(payload) 
+                => self.build_leader_block_commit_tx(payload, op_signer),
+            BurnchainOperationType::LeaderKeyRegister(payload) 
+                => self.build_leader_key_register_tx(payload, op_signer),
+            BurnchainOperationType::UserBurnSupport(payload) 
+                => self.build_user_burn_support_tx(payload, op_signer)
+        };
+
+        let transaction = match transaction {
+            Some(tx) => SerializedTx::new(tx),
+            _ => return
+        };
+
+        self.send_transaction(transaction);
+    }
+    
+    #[cfg(test)]
+    fn bootstrap_chain(&mut self) {
+
+        if let Some(local_mining_pubkey) = &self.config.burnchain.local_mining_public_key {
+
+            let pk = hex_bytes(&local_mining_pubkey).expect("Invalid byte sequence");
+            let pkh = Hash160::from_data(&pk).to_bytes().to_vec();
+            let address = BitcoinAddress::from_bytes(
+                BitcoinNetworkType::Regtest,
+                BitcoinAddressType::PublicKeyHash,
+                &pkh)
+                .expect("Public key incorrect");
+
+            let request_builder = self.get_rpc_request_builder();
+
+            let _result = BitcoinRPCRequest::import_public_key(
+                request_builder, 
+                local_mining_pubkey);
+    
+            let request_builder = self.get_rpc_request_builder();
+            let result = BitcoinRPCRequest::generate_to_address(
+                request_builder, 
+                201,
+                address.to_b58());
+
+            match result {
+                Ok(_) => {},
+                Err(e) => {
+                    error!("Bitcoin RPC failure: error generating block {:?}", e);
+                    panic!();
+                }
+            }
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+struct SerializedTx {
+    bytes: Vec<u8>
+}
+
+impl SerializedTx {
+    pub fn new(tx: Transaction) -> SerializedTx {
+        let mut encoder = RawEncoder::new(Cursor::new(vec![]));
+        tx.consensus_encode(&mut encoder).expect("BUG: failed to serialize to a vec");
+        let bytes: Vec<u8> = encoder.into_inner().into_inner(); 
+        SerializedTx {
+            bytes
+        }
+    }
+
+    fn to_hex(self) -> String {
+        let formatted_bytes: Vec<String> = self.bytes.iter().map(|b| format!("{:02x}", b)).collect();
+        format!("{}", formatted_bytes.join(""))
+    }
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct UTXO {
+    pub txid: String,
+    pub vout: u32,
+    pub script_pub_key: String,
+    pub amount: f64,
+    pub confirmations: u32,
+    pub spendable: bool,
+    pub solvable: bool,
+    pub desc: Option<String>,
+    pub safe: bool,
+}
+
+impl UTXO {
+
```

Comment:
With getters for `txid` and `script_pub_key`, should these respective fields be private?

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1381/commits/84b4605fa42b8f5111321e76e6cce380de007d52